python-main.js

```diff
@@ -1340,19 +1329,16 @@ function web_editor(config) {
         console.log(err);
         console.trace();
 
-        // If there was an error and quick flash is in use, then clear dapwrapper
-        if(usePartialFlashing) {
-            if(window.dapwrapper) {
+        // Disconnect from the microbit
+        doDisconnect().then(function() {
+            // As there has been an error clear the partial flashing DAPWrapper
+            if (window.dapwrapper) {
                 window.dapwrapper = null;
             }
-
-            if(window.previousDapWrapper) {
+            if (window.previousDapWrapper) {
                 window.previousDapWrapper = null;
             }
-        }
-
-        // Disconnect from the microbit
-        doDisconnect();
```

Comment:
At this point of webusbErrorHandler() we had already set to null the `window.dapwrapper` and `window.previousDapWrapper` variables above.

So doDisconnect() never attempted to execute `if (window.dapwrapper) window.dapwrapper.disconnectAsync();` since `dapwrapper` was falsy.